src/pages/BlogPostList.tsx

```diff
@@ -0,0 +1,110 @@
+import BlogPostCard from "../components/BlogPostCard";
+import { LayoutWrapper } from "../components/LayoutWrapper";
+import {
+  Pagination,
+  PaginationContent,
+  PaginationItem,
+  PaginationPrevious,
+  PaginationNext,
+  PaginationLink,
+  PaginationEllipsis
+} from "../components/ui/Pagination";
+import { useBlog } from "../context/BlogContext";
+import { usePagination } from "../hooks/usePagination";
+
+interface BlogPostListProps {
+  postsPerPage?: number;
+}
+
+const BlogPostList: React.FC<BlogPostListProps> = ({ postsPerPage = 6 }) => {
+  const { filteredPosts } = useBlog();
+  const {
+    currentPage,
+    totalPages,
+    goToNextPage,
+    goToPreviousPage,
+    goToPage,
+    getVisiblePages
+  } = usePagination(filteredPosts.length, postsPerPage);
+
+  const paginatedPosts = filteredPosts.slice(
+    (currentPage - 1) * postsPerPage,
+    currentPage * postsPerPage
+  );
```

Comment:
The behavior of `paginatedPosts` is puzzling. 
When you delete a post, the number of posts in the list becomes fewer than 6.
I am not sure where the issue is, or maybe it was intended this way? 😛